module/VuFind/src/VuFind/ILS/Driver/Demo.php

```diff
@@ -1078,6 +1079,7 @@ protected function getRandomTransactionList()
                         = $this->getRandomBibIdAndTitle();
                     $transList[$i]['source'] = $this->getRecordSource();
                 } else {
+                    $transList[$i]['id'] = "demo_nonexistent_$i";
```

Comment:
The purpose of the idsInMyResearch property checked above is explicitly to exclude ids when it is set to false, so it seems wrong to add an id in this case. I honestly can't remember what use case idsInMyResearch was originally created for, so maybe we need to have a discussion about its usefulness, but I think that's a separate concern. I can dig in and refresh my memory on this topic as needed!